### ISSUE TITLE
[BUG] [Formatter] formatting not disabled inside {# djlint:off #} inside script tag

### ISSUE THREAD

**Issue by thacoon** [djlint/djLint#868]
- [x] I'm on the [latest version](https://pypi.org/project/djlint/) of djLint
- [x] I've searched the [issues](https://github.com/djlint/djLint/issues)
- [x] I've read the [docs](https://djlint.com)

## System Info
 - OS: macOS Sonoma
 - Python Version 3.11.9
 - djLint Version 1.34.1
 - template language: django default templating language

## Issue

I am sometimes using django templating language inside the script tag. When I reformat the code this gets reformatted as well, so I want to ignore that part using `{# djlint:off #}`. However this is ignored by djlint and it tries to reformat the code resulting in broken javascript code, event `{# djlint:off #}` gets reformatted as well. Currently I can only ignore the whole script tag, then nothing gets reformated inside the script tag, but as soon as I only want to ignore a specific part inside it, everything gets reformatted, which then also breaks my JS code.

A similar issue was reported, which was fixed in version 1.0.0 #166, however in that previous error the `{# djlint:off #}` comments itself have not been affected, which is now the case.

## How to Reproduce
```html
<script>
    {# djlint:off #}
    const _steps = {
      {% for step in steps %}
        '{{ step.id }}': true
        {% if not forloop.last %},{% endif %}
      {% endfor %}
    };
    {# djlint:on #}

    document.addEventListener('alpine:init', () => {
        Alpine.data('steps-table', () => {
            return {
               steps: _steps, 
            };
        });
    });
</script>
```

I run `djlint --reformat --format-js server/templates/` and get:

```html
<script>
    {
        # djlint: off #
    }
    const _steps = {
        {
            %
            for step in steps %
        }
        '{{ step.id }}': true {
            %
            if not forloop.last %
        },
        {
            % endif %
        } {
            % endfor %
        }
    };
    {
        # djlint: on #
    }
    document.addEventListener('alpine:init', () => {
        Alpine.data('steps-table', () => {
            return {
                steps: _steps,
            };
        });
    });
</script>
```

**Comment by WellingtonNico** [comment]
up, same problem here:

before:
`<script>

       {# djlint:off #}
       {% if not open_form %}
	    console.log('my code that should not be formatted')
       {% endif %}
       {# djlint:on #}

    function my_function_that_should_be_indented() {
	     $("#client_form").show();
             $("#list_users").hide();
    }
 </script>
`
    
    
 after:
`<script>
    {
      # djlint: off #
    } {
      %
      if not open_form %
    }
    console.log('my code that should not be formatted') {
      % endif %
    } {
      # djlint: on #
    }

    function my_function_that_should_be_indented() {
       $("#client_form").show();
       $("#list_users").hide();
    }
 </script>
 `

**Comment by monosans** [comment]
<https://github.com/djlint/djLint/issues/931#issuecomment-2439661445>

**Comment by gsnider2195** [comment]
> [#931 (comment)](https://github.com/djlint/djLint/issues/931#issuecomment-2439661445)

I'm seeing another issue with `djlint:off` even with `format_js=false` and `format_css=false`:

```html
{# djlint:off #}
{% block javascript %}
    <script>
        const someJavascript = "foo";
    </script>
{% endblock %}
{# djlint:on #}
```

```
Reformatting 16/16 files ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 00:00    


test_djlint.html
────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────
@@ -2,6 +2,6 @@

 {% block javascript %}
     <script>
         const someJavascript = "foo";
-    </script>
+</script>
 {% endblock %}
 {# djlint:on #}
```

I would expect this linter to not touch anything between the `djlint:off` and `djlint:on` directives.
